test/test.go

```diff
@@ -524,9 +543,18 @@ func (h *Helper) Exist(path string) bool {
 
 // MustNotExist fails if path exists.
 func (h *Helper) MustNotExist(path string) {
+	if err := h.ShouldNotExist(path); err != nil {
+		h.t.Fatalf("%+v", err)
+	}
+}
+
+// ShouldNotExist returns an error if path exists.
+func (h *Helper) ShouldNotExist(path string) error {
```

Comment:
This came from the [discussion in #vendor](https://gophers.slack.com/archives/vendor/p1487854387001915) on how to improve the tests and their output when a test fails. The suggestions which resonated with me was to _not fail a test from a helper when it is a condition  that the test was directing verifying_, and to return an error instead. On the flip side, if the error is not something that a test is explicitly testing (e.g. some unrelated problem deep in the test setup), then failing with a stack trace is fine.

For example, consider a test that is asserting that the lock file doesn't exist:

* When the helper fails the test, a more generic error message is output along with a stack trace. Then you must walk back through the stack trace to figure out which line in the test failed.

    ```
    $ go test -run TestStuff
    --- FAIL: TestStuff (0.00s)
    	test.go:64: lock.json should not exist
    		github.com/golang/dep.TestStuff
    			/Users/caro8994/src/github.com/golang/dep/txn_writer_test.go:21
    		testing.tRunner
    			/usr/local/Cellar/go/1.8/libexec/src/testing/testing.go:657
    		runtime.goexit
    			/usr/local/Cellar/go/1.8/libexec/src/runtime/asm_amd64.s:2197
    ```
* When the helper returns an error that is checked by the test, the error message and offending line in the test is more clear.

    ````
    --- FAIL: TestStuff (0.00s)
    	txn_writer_test.go:441: The lock file should not be written during a dry-run
    ````

My preference is to export and use the functions that return an error (what I was naming `Should*`) and limit the use of the original `Must*` functions which fail the test from the helper.

Let me know if the problem is ambiguous name of the function, or if it is the testing pattern (returning errors) and I will update the PR accordingly.

Again, my apologies for putting this test refactoring in the PR. At the time I asked you about whether or not it should be split out, I think we both underestimated how much of a change it would be. 😃